src/core/meshcop/meshcop_tlvs.hpp

```diff
@@ -2354,6 +2355,108 @@ class DiscoveryResponseTlv : public Tlv
     uint8_t mReserved;
 } OT_TOOL_PACKED_END;
 
+/**
+ * This class implements Joiner Advertisement TLV generation and parsing.
+ *
+ */
+OT_TOOL_PACKED_BEGIN
+class JoinerAdvertisementTlv : public Tlv
```

Comment:
If we have more than one OUI Adv data, do we want to append multiple `JoinerAdvertisementTlv`s (each with its own OUI and data), or a single `JoinerAdvertisementTlv` with multiple entries in it?

I would suggest going for multiple `JoinerAdvertisementTlv`. 

Thoughts?

Reply:
I'm not thinking to support either one in this initial implementation. Maybe we can add it later when there is a use case.

Reply:
The  main reason I bring it up is that the TLV format would impact this. And I don't think we want to change the TLV format later on.

If we want the multiple entries in same TLV, then TLV format needs to track the length of each data portion. If we go for multiple TLVs then TLV format can omit the length (derive it from TLV len itself as suggested by @simonlingoogle).

Reply:
Fair point Abtin. I actually prefer to go for multiple TLVs if we want to support the mentioned use case later (multiple OUIs/AdvData on a single deivce). Agree that we can omit the `mAdvDataLength` from the TLV implementation, I've updated the code to include this change. Please take a look.

Reply:
If we are going to support multiple data, maybe we can borrow the idea in BLE's advertisement. We don't use OUI, but service type. And users should request the service type in Thread group for new type of data. Thoughts?

Reply:
Hi Yakun, I think we had some discussions on "OUI" v.s. "Type" and have reached to an agreement on using OUI. The main reason is that "While OUI is 3 bytes, it alleviates the need for Thread Group to maintain a number authority.".